src/components/celestialmap/InteractionCanvas.vue

```diff
@@ -0,0 +1,194 @@
+<template>
+    <canvas ref="canvas" 
+        :id="canvas_id" 
+        :width="width+'px'" :height="height+'px'" 
+        :class="{ 'mouse-in-sky': mouse_in_sky }"
+        class="interaction-layer-canvas"
+        />
+</template>
+
+<script>
+export default {
+    name: "InteractionCanvas",
+    props: {
+        canvas_id: {
+            type: String,
+            default: "interaction-canvas"
+        },
+        width: {
+            type: Number,
+            default: 200
+        },
+        height: {
+            type: Number,
+            default: 200
+        },
+        user_crosshairs: {
+            type: Array,
+            default: () => [-1, -1], // render offscreen
+            validator: val => {
+                let correct_size = val.length == 2;
+                let all_elements_are_numbers = val.every(i => {
+                    return typeof i === "number"
+                })
+                return correct_size && all_elements_are_numbers;
+            },
+        },
+        telescope_crosshairs: {
+            type: Array,
+            default: () => [-1, -1], // render offscreen
+            validator: val => {
+                let correct_size = val.length == 2;
+                let all_elements_are_numbers = val.every(i => {
+                    return typeof i === "number"
+                })
+                return correct_size && all_elements_are_numbers;
+            },
+        },
+        // This changes the cursor style if the mouse is in the sky 'circle'. 
+        mouse_in_sky: {
+            type: Boolean,
+            default: false
+        }
+    },
+
+    mounted() {
+        this.canvas = this.$refs.canvas
+        this.ctx = this.canvas.getContext('2d')
+        let that = this
+
+        // Send click / drag events to parent component
```

Comment:
Interesting note. First of all, the localhost version usually runs slower than a deployed version. 

That being said, you did highlight a change I hadn't noticed: the aladin viewer updates continuously now when the user drags the crosshairs across the map. Previously the aladin viewer only updated when the user released the mouse during a drag. I think more updating is probably better, though it might be too slow to be useful and too 'flashy' as to distract the user.